oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/AuthorizationRequestRedirectFilter.java

```diff
@@ -131,6 +132,8 @@ private void sendRedirectForAuthorization(HttpServletRequest request, HttpServle
 		Map<String,Object> additionalParameters = new HashMap<>();
 		additionalParameters.put(OAuth2Parameter.REGISTRATION_ID, clientRegistration.getRegistrationId());
 
+		String nonce = request.getParameter(OAuth2Parameter.NONCE);
```

Comment:
The `nonce` does not come from the `HttpServletRequest`. It must be produced by this filter and added as a request parameter as part of the [Authentication Request](http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).

It then needs to get correlated/validated in the [ID Token](http://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation)

See [comment](https://github.com/spring-projects/spring-security/issues/4442#issue-241345744) for implementation notes

Reply:
@jgrandja I will work on these and commit